src/ResourceManager/Compute/Commands.Compute/Extension/Chef/SetAzureVMChefExtension.cs

```diff
@@ -119,9 +119,10 @@ public string TypeHandlerVersion
         [Parameter(
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "Configures the chef-client service for unattended execution. The node platform should be Windows." +
-                          "Options: 'none' or 'service'." +
```

Comment:
@NimishaS can we get rid of lot of code that is written again for null checking in ValidateParameters function, almost all the parameters are marked as
[ValidateNotNullOrEmpty]
consider using parameter sets which is really good to manage your long list of parameters.
Also consider using default value attributes if applicable for any of the parameters. That will also reduce code that is written to set defaults.